### PR TITLE
Add govuk lint

### PR DESCRIPTION
**Pull request by davidbasalla**
This adds the ruby and sass linters to the project and to jenkins.sh. The linter has been disabled for certain style violations that result from outdated code and which will be addressed in future as a part of a frontend overhaul.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -249,3 +272,6 @@ DEPENDENCIES
   uglifier (= 2.7.2)
   unicorn (~> 4.9.0)
   webmock (= 1.21.0)
+
```
Comment (davidbasalla) [inline Gemfile.lock]: Is this okay here? Or better to be removed?

**Comment by h-lame** [comment]
This is gonna need rebasing now we've merged #111.

File: Gemfile.lock (hunk 1 above)
Comment (h-lame) [inline Gemfile.lock]: It'll be fine, but once rebased make sure it's still the correct version for ruby 2.3.0.

**Comment by h-lame** [comment]
Looks good to me though, might be worth getting someone with more frontend chops to have a look at the solution chosen to fix the lint warnings in the final commit though.

**Comment by dsingleton** [comment]
Most of the violation fixes are :+1:

For the violations requiring bigger changes (eg, markup), i'd suggest using [SASS Lint comments to ignore](https://github.com/brigade/scss-lint/#disabling-linters-via-source) them instead of making the change. 

That said, I don't think the fixes here are bad, just reach a bit further. Things like renaming the font-stack make sense within this repo, but the way it handles typography is already weird/exceptional vs the rest of GOV.UK (eg, not using the FET versions of the font variable, or typography mixins). It feels simpler to highlight the current use as unusual/exceptional, but ignore it from linting, so it can be cleaned properly in the future.

It's up to you, i'd be OK merging this as-is (if we make the note that the repo needs some serious FE modernisation), but given the historic nature of the SCSS, i think it might be better to not try and update, even for the linting.

**Comment by davidbasalla** [comment]
Thanks @dsingleton, agreed that it makes sense to disable the linter and mark as needing more work in future

**Comment by davidbasalla** [comment]
@dsingleton I've added lines to ignore the more complex cases of SASS violations in the last commit. Does this look reasonable to you?

File: app/assets/stylesheets/styleguide/_font-stacks.scss [hunk 2]
```diff
@@ -32,6 +33,7 @@ $NTA-Light:
        local("Helvetica");
 }
 
+// scss-lint:disable NameFormat
```
Comment (fofr) [inline app/assets/stylesheets/styleguide/_font-stacks.scss]: Are these variables being used? Shouldn't they be pulled from the frontend toolkit? Probably a question for a different PR.

File: app/assets/stylesheets/styleguide/_typography.scss [hunk 3]
```diff
@@ -36,6 +36,7 @@
 // CORE FONTS - NEW TRANSPORT
 
 @mixin core-48($line-height: (50 / 48), $line-height-640: (30 / 25)) {
+  // scss-lint:disable NameFormat
```
Comment (fofr) [inline app/assets/stylesheets/styleguide/_typography.scss]: I'd expect this file to be redundant given what's in frontend toolkit. Looks like there's a piece of work to update this app to _the modern way_.

Comment (davidbasalla) [inline app/assets/stylesheets/styleguide/_typography.scss]: Yeah, chatted with @dsingleton about this and concluded that this app would need some serious frontend modernising at some point!

**Comment by fofr** [comment]
Changes LGTM :+1: :heart:

**Pull request merged**